typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +150,541 @@ export class EvmIsmModule extends HyperlaneModule<
+      config.domains,
+      (domain, config): config is IsmConfig => {
+        const domainId = multiProvider.tryGetDomainId(domain);
+        if (domainId === null) {
+          logger.warn(
+            `Domain ${domain} doesn't have chain metadata provided, skipping ...`,
+          );
+        }
+        return domainId !== null;
+      },
+    );
+
+    const safeConfigDomains = Object.keys(config.domains).map((domain) =>
+      multiProvider.getDomainId(domain),
+    );
+
+    const isms: ChainMap<Address> = {};
+    const owner = await resolveOrDeployAccountOwner(
+      multiProvider,
+      destination,
+      config.owner,
+    );
+
+    for (const origin of Object.keys(config.domains)) {
+      const ism = await EvmIsmModule.deploy({
+        destination,
+        config: config.domains[origin],
+        origin,
+        mailbox,
+        multiProvider,
+        factories,
+        deployer,
+      });
+      isms[origin] = ism.address;
+    }
+
+    const submoduleAddresses = Object.values(isms);
+
+    if (config.type === IsmType.FALLBACK_ROUTING) {
+      // deploying new fallback routing ISM
+      if (!mailbox) {
+        throw new Error(
+          'Mailbox address is required for deploying fallback routing ISM',
+        );
+      }
+      logger.debug('Deploying fallback routing ISM ...');
+      routingIsm = await multiProvider.handleDeploy(
+        destination,
+        new DefaultFallbackRoutingIsm__factory(),
+        [mailbox],
+      );
+    } else {
+      routingIsm = await EvmIsmModule.deployDomainRoutingIsm({
+        destination,
+        owner,
+        safeConfigDomains,
+        submoduleAddresses,
+        overrides,
+        multiProvider,
+        factories,
+      });
+    }
+
+    return routingIsm;
+  }
+
+  protected static async deployDomainRoutingIsm(params: {
+    destination: ChainName;
+    owner: string;
+    safeConfigDomains: number[];
+    submoduleAddresses: string[];
+    overrides: ethers.Overrides;
+    multiProvider: MultiProvider;
+    factories: HyperlaneContracts<ProxyFactoryFactories>;
+  }): Promise<DomainRoutingIsm> {
+    const {
+      destination,
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      overrides,
+      multiProvider,
+      factories,
+    } = params;
+
+    // deploying new domain routing ISM
+    const tx = await factories.domainRoutingIsmFactory.deploy(
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      overrides,
+    );
+
+    const receipt = await multiProvider.handleTx(destination, tx);
+
+    // TODO: Break this out into a generalized function
+    const dispatchLogs = receipt.logs
+      .map((log) => {
+        try {
+          return factories.domainRoutingIsmFactory.interface.parseLog(log);
+        } catch (e) {
+          return undefined;
+        }
+      })
+      .filter(
+        (log): log is ethers.utils.LogDescription =>
+          !!log && log.name === 'ModuleDeployed',
+      );
```

Comment:
I see you copied this line for line (including the TODO) from HyperlaneIsmFactory. Could we DRY this up or is the plan to kill HyperlaneIsmFactory soon anyway?

Reply:
i thought about DRYing it up but as you guessed, yeah I want to get to a state where we can nuke the factory/deployers in favour of these modules